src/Uno.UI/UI/Xaml/Internal/RootVisual.cs

```diff
@@ -33,18 +33,14 @@ namespace Uno.UI.Xaml.Core;
 internal partial class RootVisual : Panel, IRootElement
 {
 	private readonly CoreServices _coreServices;
-	private readonly UnoRootElementLogic _rootElementLogic;
 
 	public RootVisual(CoreServices coreServices)
 	{
 		_coreServices = coreServices ?? throw new System.ArgumentNullException(nameof(coreServices));
-		_rootElementLogic = new(this);
-	}
-
-	void IRootElement.NotifyFocusChanged() => _rootElementLogic.NotifyFocusChanged();
 
-	void IRootElement.ProcessPointerUp(PointerRoutedEventArgs args, bool isAfterHandledUp) =>
-		_rootElementLogic.ProcessPointerUp(args, isAfterHandledUp);
+		//Uno specific - flag as VisualTreeRoot for interop with existing logic
+		IsVisualTreeRoot = true;
```

Comment:
Is this needed? I thought the information is retrievable in other ways

Reply:
Yes it's being used in few hot-path code to avoid cast, like EVP. But indeed having this in all `UIElement`  is maybe actually more expensive ... we can review this, but won't do it in current PR :)